lib/ansible/parsing/mod_args.py

```diff
@@ -301,6 +301,11 @@ def parse(self, skip_action_validation=False):
                     raise AnsibleParserError("conflicting action statements: %s, %s" % (action, item), obj=self._task_ds)
                 action = item
                 thing = value
+
+                context = module_loader.find_plugin_with_context(item, check_aliases=True, collection_list=self._collection_list)
+                if context.resolved and context.redirect_list:
+                    self._task_ds['ansible_internal_redirect_list'] = context.redirect_list
```

Comment:
check w/ @jimi-c on this one- it definitely works, but dunno if there's a better way or if there are risks to this approach I'm unaware of...

Reply:
Improved this in c5853621fb53c30b8ff1368bd1070c9515c2511b

Thanks @bcoca and @jimi-c for the help!